drivers/dma/bcm2708-dmaengine.c

```diff
@@ -734,8 +734,6 @@ static int bcm2835_dma_chan_init(struct bcm2835_dmadev *d, int chan_id, int irq)
 	vchan_init(&c->vc, &d->ddev);
 	INIT_LIST_HEAD(&c->node);
 
-	d->ddev.chancnt++;
```

Comment:
Can you explain why this change is necessary (or why the old code is no longer necessary)?